solutions/gold/cf-1462E2.mdx

```diff
@@ -101,4 +101,88 @@ int main() {
 }
 ```
 </CPPSection>
+
+<PySection>
+
+```py
+MAXN = 200001
+MOD = 10**9 + 7
+
+fac = [0] * (MAXN + 1)
+inv = [0] * (MAXN + 1)
+
+# BeginCodeSnip{Combinatorics Functions (from the module)}
+def exp(x: int, n: int, m: int) -> int:
+	x %= m
+	res = 1
+	while n > 0:
+		if n % 2 == 1:
+			res = res * x % m
+		x = x * x % m
+		n //= 2
+	return res
+
+
+def factorial():
+	fac[0] = 1
+	for i in range(1, MAXN + 1):
+		fac[i] = fac[i - 1] * i % MOD
+
+
+def inverses():
+	inv[MAXN] = exp(fac[MAXN], MOD - 2, MOD)
+	for i in range(MAXN, 0, -1):
+		inv[i - 1] = inv[i] * i % MOD
+
+
+def choose(n: int, r: int):
+	if n < r:
+		return 0
+	return fac[n] * inv[r] % MOD * inv[n - r] % MOD
+
+
+# EndCodeSnip
+
+factorial()
+inverses()
+for _ in range(int(input())):
+
+	n, m, k = map(int, input().split())
+	nums = list(map(int, input().split()))
+	freq = [0] * (n + 1)  # frequency of each element
+	preFreq = [0] * (n + 1)  # prefix-array for freq vector
+	ans = 0
+
+	for x in nums:
+		freq[x] += 1
+
+	for i in range(1, n + 1):
+		preFreq[i] = freq[i] + preFreq[i - 1]
```

Comment:
why camelcase for this name bro

Reply:
i was just keeping the var names same, which you only use to tell me

Reply:
should i use snake case here

Reply:
yeah
idk what the c++ version is on...

Reply:
snake case implemented